### ISSUE TITLE
wine64: wine64-preloader crashes intermittently on macOS 13.3

### ISSUE THREAD

**Issue by reckenrode** [NixOS/nixpkgs#224209]
### Describe the bug

Changes in macOS 13 caused the Wine preloader (`wine64-preloader`) to issue a warning when Wine was run on that version of macOS. Up until macOS 13.3, the warning was mostly harmless. As of macOS 13.3, the preloader can crash instead of issuing the warning. This issue is fixed in Wine 8.5 and will be fixed in 8.0.1 (which is not available yet).

I can create a PR to bump Wine to 8.5, which will fix the issue on the unstable release. I also want to do one for 8.0.1 once that is released. Given that 23.05 is still a little ways off, waiting on the release seems preferable to temporarily backporting the patches then bumping it anyway.

### Steps To Reproduce

1. Upgrade a macOS system to 13.3
2. Run `WINEPREFIX=<some prefix> wineboot --init`
3. Observe the crash (probably)

### Expected behavior

Wine runs on Darwin.

### Screenshots

When it fails, the failures vary. Sometimes Wine can’t call `CertFreeCertificateContext` in `crypt32`. Other times it can’t get the display config. I have a derivation that creates a Wine prefix in the Nix store, which usually crashes with a `SIGBUS`.

Naturally, when I want to reproduce the crash, it works instead, so I’ve included an excerpt from the `SIGBUS` crash because that was logged by macOS. The memory range identified is the same one that was generating a warning in prior releases of macOS 13.3.

```console
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               wine64-preloader [16143]
Path:                  /Volumes/VOLUME/*/wine64-preloader
Identifier:            wine64-preloader
Version:               ???
Code Type:             X86-64 (Translated)
Parent Process:        Exited process [16134]
Responsible:           nix [780]
User ID:               301

Date/Time:             2023-03-29 10:02:40.8973 -0400
OS Version:            macOS 13.3 (22E252)
Report Version:        12
Anonymous UUID:        AE134603-5851-8277-863D-304B4C70126C

Sleep/Wake UUID:       91FE54D4-9944-4E58-8D70-AB521184A98F

Time Awake Since Boot: 3500 seconds

System Integrity Protection: enabled

Notes:
dyld_process_snapshot_get_shared_cache failed

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGBUS)
Exception Codes:       KERN_PROTECTION_FAILURE at 0x0000000000010000
Exception Codes:       0x0000000000000002, 0x0000000000010000

Termination Reason:    Namespace SIGNAL, Code 10 Bus error: 10
Terminating Process:   exc handler [16143]

VM Region Info: 0x10000 is in 0x10000-0x110000;  bytes after start: 0  bytes before end: 1048575
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      dyld private memory              1000-10000        [   60K] rw-/rwx SM=PRV  
--->  VM_ALLOCATE                     10000-110000       [ 1024K] ---/rwx SM=NUL  
      VM_ALLOCATE                    110000-8110000      [128.0M] ---/rwx SM=NUL  
```

### Additional context

Upstream bug: https://bugs.winehq.org/show_bug.cgi?id=54420

### Notify maintainers

@avnik @7c6f434c @bendlas @jmc-figueira

Actually, I wonder if I should also add myself to the maintainers list since I’m actively using Wine on Darwin and want to keep it working (and have submitted several PRs to effect that).

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.4.0, macOS 13.3`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

**Comment by 7c6f434c** [comment]
Yeah, absolutely, if you constantly use it on Darwin, add yourself to the maintainer list.

**Comment by phanirithvij** [comment]
Closing this because the linked pr was merged, re-open if it should still be open.

**Comment by reckenrode** [comment]
Wine no longer uses the preloader on Darwin, so it’s unlikely this particular issue will occur again in the future. It instead uses a linker feature (supported by the version of ld64 in nixpkgs) to accomplish the same goal.
